systems/Query/05-Building_Block_View.md

```diff
@@ -0,0 +1,67 @@
+| --------------------------------- | ---------------------------------------------------------------------- |
+| [Web Interfaces](#web-interfaces) | Allow user interaction with the Query Service via UIs                  |
+| [Backend](#backend)               | Index data and serve [SPARQL](./../../Glossary.md#sparql) queries      |
+| [Updating](#updating)             | Keeping the Backend up to date with changes from a Wikibase Repository |
+
+## Web Interfaces
+
+![](./diagrams/05-01-web-interfaces.drawio.svg)
+
+| Building Block | Responsibility                                                                                                              |
+| -------------- | --------------------------------------------------------------------------------------------------------------------------- |
+| Query Frontend | Provide users with an interface to create entirly custom [SPARQL](./../../Glossary.md#sparql) queries and visualize results |
+| Query Builder  | Provider users with a simple visual interface for generating lists using the query service                                  |
+
+## Backend
+
+![](./diagrams/05-01-backend.drawio.svg)
+
+| Building Block | Responsibility                                                  |
+| -------------- | --------------------------------------------------------------- |
+| Query Proxy    | Force setting of timeouts, read only queries and basic caching  |
+| Blazegraph     | Graph datagbase with a [SPARQL](./../../Glossary.md#sparql) API |
+| Data Loader    | Loads data into Blazegraph from an RDF rump                     |
+
+### Data Loader
+
+The Data Loader currently currently exists in 2 different flavours, which are used in difference deployments.
+Only 1 of the choices for munging / RDF conversaion is needed per deployment.
+
+![](./diagrams/05-02-backend-loader.drawio.svg)
+
+| Building Block      | Responsibility                                                                                                                                             |
+| ------------------- | ---------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| loadData            | Load [RDF](./../../Glossary.md#rdf) into Blazegraph                                                                                                        |
+| munger              | Convert Wikibase Repository [RDF](./../../Glossary.md#rdf) triples into different triples for Blazegraph                                                   |
+| TurtleDumpConverter | Convert Wikibase Repository [RDF](./../../Glossary.md#rdf) triples into different triples for Blazegraph in a [Hadoop](./../../Glossary.md#hadoop) cluster |
+
+## Updating
+
+The Updating system curently exists in 3 different flavours, which are used in different deployments.
+Only 1 of these update paths is needed in each deployment.
+
+![](./diagrams/05-01-updating.drawio.svg)
```

Comment:
> It would be nice if this diagram could somehow communicate that some of these flavours pull their data from Wikibase, wheras others get it pushed to them.

This is probably something better represented in the sequence diagrams, as at this level we only show data flow.
A sequence diagram for the streaming updater will probably come after this initial PR gets merged.

> And generally what the difference between the top two ones is. Besides the name, they seem almost identical

The difference here is detailed in the table just below the diagram.

Reply:
"Only one of these update paths is needed", but only two of the components are marked as optional and the RCPoller must always be there? If that's the case, then maybe the RCPoller deserves one more sentence in the intro paragraph above the chart. Apparently it is also the default choice for the Wikibase Docker deployment.

Reply:
Good point.
Indeed, the RCPoller is more the default updater, for wikibase deployments at least.
Wikidata uses the streaming updater, and the other kafka updater is not really used at all right now afaik (probably pending removal).
Thanks for the note, I'll updater something here